### PR TITLE
Joomla CMS [#27415] Missing platform change for update notification

### PR DESCRIPTION
**Pull request by dextercowley**
I missed a platform change when this feature was added to the CMS. This is the change. It is already in the CMS version of this file. Thanks.

### CODE REVIEW

File: libraries/joomla/updater/updater.php [hunk 1]
```diff
@@ -63,12 +63,13 @@ public static function &getInstance()
 	 * Finds an update for an extension
 	 *
 	 * @param   integer  $eid  Extension Identifier; if zero use all sites
```
Comment (realityking) [inline libraries/joomla/updater/updater.php]: This needs to align with the param below it.

File: libraries/joomla/updater/updater.php [hunk 2]
```diff
@@ -63,12 +63,13 @@ public static function &getInstance()
...
+	 * @param   integer  $cache_timeout  How many seconds to cache update information; if zero, force reload the update information
 	 *
 	 * @return  boolean True if there are updates
 	 *
 	 * @since   11.1
 	 */
-	public function findUpdates($eid = 0)
+	public function findUpdates($eid=0, $cache_timeout = 0)
```
Comment (realityking) [inline libraries/joomla/updater/updater.php]: The equal sign needs to be surrounded by spaces.

File: libraries/joomla/updater/updater.php [hunk 3]
```diff
@@ -103,6 +105,14 @@ public function findUpdates($eid = 0)
 				// Ignore update sites requiring adapters we don't have installed
 				continue;
 			}
+			if($cache_timeout > 0) {
```
Comment (realityking) [inline libraries/joomla/updater/updater.php]: Missing space behind the if, opening parenthesis must be on a new line.

File: libraries/joomla/updater/updater.php [hunk 4]
```diff
@@ -103,6 +105,14 @@ public function findUpdates($eid = 0)
...
+				if ($now - $result['last_check_timestamp'] <= $cache_timeout) {
```
Comment (realityking) [inline libraries/joomla/updater/updater.php]: Opening parenthesis must be on a new line.

File: libraries/joomla/updater/updater.php [hunk 5]
```diff
@@ -172,6 +182,14 @@ public function findUpdates($eid = 0)
 			{
 				$update_result = true;
 			}
+			
+			// Finally, update the last update check timestamp
+			$query = $dbo->getQuery(true);
+			$query->update($dbo->quoteName('#__update_sites'));
+			$query->set($dbo->quoteName('last_check_timestamp').' = '.$dbo->quote($now));
```
Comment (realityking) [inline libraries/joomla/updater/updater.php]: When concating string the . operator must be surrounded by spaces.

File: libraries/joomla/updater/updater.php [hunk 6]
```diff
@@ -172,6 +182,14 @@ public function findUpdates($eid = 0)
...
+			$query->where($dbo->quoteName('update_site_id').' = '.$dbo->quote($result['update_site_id']));
```
Comment (realityking) [inline libraries/joomla/updater/updater.php]: Same as above.

**Comment by elkuku** [comment]
There are some more... http://elkuku.github.com/pulltester/pulls/657

File: libraries/joomla/updater/updater.php (hunk 2 above)
Comment (LouisLandry) [inline libraries/joomla/updater/updater.php]: I'd also like to see the $cache_timeout changed to $cacheTimeout.  We should be using camelcase for everything.

**Comment by elkuku** [comment]
Almost ;) three new ones.. Tip: Eclipse save actions (PHP): remove trailing white space ;)
http://elkuku.github.com/pulltester/pulls/657

**Comment by dextercowley** [comment]
Thanks for the Eclipse tip! Mark

On Mon, Dec 19, 2011 at 5:18 PM, Nikolai Plath
reply@reply.github.com
wrote:

**Pull request merged**